File: 89/89.go
func grayCode(n int) []int {
	nn := 1 << n
	// int mask = nn-1;
	res := make([]int, nn)

	for i := 0; i < nn; i++ {
		res[i] = i ^ (i >> 1)
	}
	return res
}

File: 89/89.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        int nn = 1<<n;
        // int mask = nn-1;
        vector<int> res (nn);
        
        for (int i = 0; i < nn; i++){
            res[i] = i^(i>>1) ;
        }
        return res;
    }
};